File: unit-3/sprint-3/day-1/we/servicewepart/app/src/main/java/com/example/servicewepart/MainActivity.java
package com.example.servicewepart;

import androidx.appcompat.app.AppCompatActivity;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private EditText mEtText;
    private Button mBtnButton;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
        IntentFilter intentFilter = new IntentFilter("com.abc.service");
        registerReceiver(serviceReceiver,intentFilter);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        unregisterReceiver(serviceReceiver);
    }

    private void initView() {
        mEtText = findViewById(R.id.etEditText);
        mBtnButton = findViewById(R.id.btnButton);
        mBtnButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this,FileOperationService.class);
                intent.putExtra("name",mEtText.getText().toString());
                startService(intent);
            }
        });
    }

    private BroadcastReceiver serviceReceiver = new BroadcastReceiver() {
        @Override
        public void onReceive(Context context, Intent intent) {
            Toast.makeText(context,intent.getStringExtra("data"),Toast.LENGTH_SHORT).show();
        }
    };
}

File: unit-4/sprint-2/evaluation/unit4_codingeval_1/app/src/main/java/com/example/unit4_codingeval_1/MainActivity.kt
package com.example.unit4_codingeval_1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.FragmentManager

class MainActivity : AppCompatActivity() {
    lateinit var fragmentManager: FragmentManager
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        fragmentManager = supportFragmentManager
        setFragment1()
        setFragment2()
    }

    private fun setFragment1() {
        val fragmenttransaction = fragmentManager.beginTransaction()
        val fragment1 = AddItemFragment()
        fragmenttransaction.replace(R.id.flContainer1,fragment1).addToBackStack("add item frag").commit()
    }

    private fun setFragment2() {
        val fragmenttransaction = fragmentManager.beginTransaction()
        val fragment2 = SearchItemFragment()
        fragmenttransaction.replace(R.id.flContainer2,fragment2).addToBackStack("search fragment").commit()

    }


}